### PR TITLE
[Alura] New extractor

### PR DESCRIPTION
**Pull request by hugohaa**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Alura is the biggest tech learning platform in Brazil. 

This PR provides single-video and full-course download methods. It's mandatory to have a valid account to download content.

### CODE REVIEW

**Comment by blackjack4494** [comment]
added it to my fork 👍

File: youtube_dl/extractor/alura.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    _NETRC_MACHINE = 'alura'
+    _TESTS = [{
+        'url': 'https://cursos.alura.com.br/course/clojure-mutabilidade-com-atoms-e-refs/task/60095',
+        'info_dict': {
+            'id': '60095',
+            'ext': 'mp4',
+            'title': 'Referências, ref-set e alter'
+        },
+        'skip': 'Requires alura account credentials'},
+        {
```
Comment (dirkf) [inline youtube_dl/extractor/alura.py:34]: More maintainable:
```suggestion
        'skip': 'Requires alura account credentials',#
    }, {
```

File: youtube_dl/extractor/alura.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            'only_matching': True},
+        {
+            'url': 'https://cursos.alura.com.br/course/fundamentos-market-digital/task/55219',
+            'only_matching': True}
+    ]
+
+    def _real_extract(self, url):
+
+        video_id = self._match_id(url)
+        course = self._search_regex(self._VALID_URL, url, 'post url', group='course_name')
```
Comment (dirkf) [inline youtube_dl/extractor/alura.py:46]:
```suggestion
        course, video_id = re.match(self._VALID_URL, url).group('course_name', 'id')
```

**Review by dirkf — CHANGES_REQUESTED** [review]
This won't be merged unless someone with an account can test it.

**Pull request closed without merging**